### ISSUE TITLE
Weird display of Android sample: stream test

### ISSUE THREAD

**Issue by wangziji** [google/filament#2831]
I are trying to use camera preview as an external texture. It works on most devices but on some other device, its displayed weird (check pics below). I have try to run the stream test and got the same result. Our solution is similar to the texture id one.
![image](https://user-images.githubusercontent.com/3860989/87918139-961cae80-caa8-11ea-8bff-83260e11cbf0.png)

**To Reproduce**
1. Run the android stream test sample.

**Expected behavior**
1. Display normal

**Screenshots**
Vivo X7:
![image](https://user-images.githubusercontent.com/3860989/87915875-4688b380-caa5-11ea-81d0-6a2b472aac22.png)
![image](https://user-images.githubusercontent.com/3860989/87915992-7041da80-caa5-11ea-8087-6bd4807d6d24.png)

Vivo X9Plus L:
![image](https://user-images.githubusercontent.com/3860989/87916250-c57dec00-caa5-11ea-9994-2d3f48be588e.png)
![image](https://user-images.githubusercontent.com/3860989/87916311-daf31600-caa5-11ea-96a1-e7a25c1cb387.png)

Oppo R9M:
![image](https://user-images.githubusercontent.com/3860989/87917598-d0398080-caa7-11ea-8eac-e1a4c9b422df.png)
![image](https://user-images.githubusercontent.com/3860989/87917295-5608fc00-caa7-11ea-8ea7-2513ee2e52fe.png)

**Android device:**
 - Device: Vivo X7 / Vivo X9Plus L / Oppo R9M / Meizu MX5
 - OS: Android 5.1.1 / Android 7.1.2 / Android 5.1 / Android 5.1
 - GPU: Adreno510 / Adreno510 /  Adreno510 / PowerVR G6200

**Comment by prideout** [comment]
Does our `sample-hello-camera` demo work on these devices?

We mostly test with Pixel 3 and Pixel 4, this might be a driver issue.

**Comment by wangziji** [comment]
> Does our `sample-hello-camera` demo work on these devices?
> 
> We mostly test with Pixel 3 and Pixel 4, this might be a driver issue.

sample-hello-camera is not working too.  But it is due to API compatibility problem.

07-21 13:49:25.212 29714-29714/com.google.android.filament.hellocamera E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.google.android.filament.hellocamera, PID: 29714
    java.lang.NoSuchMethodError: No static method newInstance(IIIIJ)Landroid/media/ImageReader; in class Landroid/media/ImageReader; or its super classes (declaration of 'android.media.ImageReader' appears in /system/framework/framework.jar)
        at com.google.android.filament.hellocam.CameraHelper.<init>(CameraHelper.kt:56)
        at com.google.android.filament.hellocam.MainActivity.onCreate(MainActivity.kt:97)
        at android.app.Activity.performCreate(Activity.java:6251)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1112)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2622)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2766)
        at android.app.ActivityThread.access$1000(ActivityThread.java:197)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1611)
        at android.os.Handler.dispatchMessage(Handler.java:111)
        at android.os.Looper.loop(Looper.java:224)
        at android.app.ActivityThread.main(ActivityThread.java:5958)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1113)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:879)

**Comment by romainguy** [comment]
This sample app requires a newer version of Android. These problems look like they are specific to those devices and versions of android. I don’t think there’s anything we can do about it on our side unfortunately.

**Comment by wangziji** [comment]
I understand the sample-hallo-camera request new Android API, but why sample-stream-test needs new API?

Our purpose is to bind camera preview to a given opengl oes texture id, then render this texture with Filament. We already successfully tested this solution with most devices which supported the hardware buffer. However, for those devices which API level under 26. We got black screen or mosaic like display.

I believe the way we can implement the camera preview is very similar to the sample-stream-test. That's why we decided to test the sample-stream-test with above devices.
